packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m

```diff
@@ -119,9 +115,38 @@ - (instancetype)initWithFrame:(CGRect)frame
 }
 
 - (UIView*)view {
+  [_mapView addObserver:self forKeyPath:@"frame" options:0 context:nil];
   return _mapView;
 }
 
+- (void)observeValueForKeyPath:(NSString*)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary*)change
+                       context:(void*)context {
+  if (_cameraDidInitialSetup) {
+    // We only observe the frame for initial setup.
+    [_mapView removeObserver:self forKeyPath:@"frame"];
+    return;
+  }
+  if (object == _mapView && [keyPath isEqualToString:@"frame"]) {
+    CGRect bounds = _mapView.bounds;
+    if (CGRectEqualToRect(bounds, CGRectZero)) {
+      // Rerely, frame can change without actually changing the size of the view;
+      // eg, consider a frame change such as: (0, 0, 0, 0) -> (10, 10, 0, 0)
```

Comment:
You mean (0, 0, 0, 0) -> (10, 10, 10, 10) ?
Is this not the change we get any time we "move" a map? (is that rare?)

I see why we can ignore these changes - this whole thing is working around an issue with our initial frame being sized to zero, but if a consequent frame is zero there's nothing to workaround really... (what I'm saying the code makes sense to me, the comment makes less sense)

Reply:
How about just:
```
      // The workaround is to fix an issue that the camera location is not current when
      // the size of the map is zero at initialization.
      // So We only care about the size of the `_mapView`, ignore the frame changes when the size is zero.
```